.github/compatibility-test-matrices/compat.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/python3
+import argparse
+import os
+import json
+
+# JSON keys to search in.
+KEYS = ("chain-a", "chain-b")
+# Toggle if required. Indent = 4 matches jq.
+DUMP_ARGS = {"indent": 4, "sort_keys": False, "ensure_ascii": False}
+# Suggestions for recent and new versions.
+SUGGESTION = "for example, v4.3.0 or 4.3.0"
+
+def find_json_files(directory, ignores=(".",)):
+    """ Find JSON files in a directory. By default, ignore hidden directories."""
+    for root, dirs, files in os.walk(directory):
+        dirs[:] = (d for d in dirs if not d.startswith(ignores))
```

Comment:
totally ugly, agree there but this in-place assignment of the returned directories allows for skipping any filtered entries when `topdown=True`. Alternative would be an iteration over the dirs and manual `del` of the correct indices.

In general, `ignores` was added based on the initial bash script considering a `.backup` dir being present. Since I haven't added that in (thinking a `git restore` could do the trick) this could also be removed).

Reply:
oh I see so we're actually modifying the dirs that we're already iterating over